src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -12,25 +12,34 @@
   private CANSparkMax intakeMotor;
   private DoubleSolenoid intakePiston;
 
+  private boolean deployed;
+
   public Intake()
   {
     intakeMotor = new CANSparkMax(Constants.INTAKE_CAN_SPARK, MotorType.kBrushless); // Motor is Brushless
     intakePiston = new DoubleSolenoid(Constants.INTAKE_PISTON_DEPLOY, Constants.INTAKE_PISTON_RETRACT);
+
+    deployed = false;
   }
 
   public void deploy()
   {
+    deployed = true;
     intakePiston.set(Value.kForward);
   }
 
   public void retract()
   {
+    deployed = false;
     intakePiston.set(Value.kReverse);
   }
 
   public void spin()
   {
-    intakeMotor.set(0.25);
+    if (deployed == true)
```

Comment:
You don't really need to write `== true` here.  It is already a `boolean` and the compiler will optimize this code out anyway.  It is much easier to read `if (deployed) { do stuff }`.